crates/burn-tensor/src/tensor/api/numeric.rs

```diff
@@ -2030,6 +2030,95 @@ where
         // Assign the original tensor data to the appropriate slice of the padded tensor
         padded_tensor.slice_assign(ranges, self)
     }
+    /// Create a one-hot encoded tensor with configurable `depth`, `on_value`, `off_value`, and `axis` including high-ranked tensors.
+    ///
+    /// # Arguments
+    ///
+    /// * `depth`: The number of classes for the one-hot encoding, which defines the size of the one-hot dimension.
```

Comment:
Not a big fan of `depth`, I think `num_classes` is more descriptive / representative.

Reply:
I agree with your comment, I rename it.